packages/web-config-server/src/apiV1/dataBuilders/generic/compose/composeDataPerDataClass.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { fetchComposedData } from '/apiV1/dataBuilders/helpers';
+
+/**
+ * Configuration schema
+ * @typedef {Object} ComposeDataConfig
+ * @property {string} sortOrder
+ * @property {Object<string, { dataBuilder, dataBuilderConfig }>} dataBuilders
+ *
+ * Example config:
+ * {
+ *   dataBuilders: {
+ *        builder1: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        },
+ *        builder2: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        }
+ *    }
+ * }
+ * 
+ * Data returned from fetchComposedData():
+ * {
+ *     builder1: { 
+ *        data: [{name: dataClass1: value: '111'}, {name: dataClass2: value: '222'}]
+ *     },
+ *     builder2: {
+ *        data: [{name: dataClass1: value: '333'}, {name: dataClass2: value: '444'}],
+ *     }
+ * }
+ * 
+ * The above will then be converted to the below and returned:
+ * { 
+ *    data: 
+ *        [
+ *          { name: dataClass1, builder1: '111', builder2: '333' }, 
+ *          { name: dataClass2, builder1: '222', builder2: '444' }, 
+ *        ]
+ * }
+ */
+
+export const composeDataPerDataClass = async (config, aggregator, dhisApi) => {
```

Comment:
@edmofro Do you care about functional data builders (which doesn't extend from DataBuilder class)?

Reply:
Sometimes! In this case, there will be no conflict, but if the functional data builder either:
- Use a model through direct import, or 
- Called another functional data builder that relied on models, and didn't pass it in (in this case `fetchComposedData` will receive the models correctly through `config`)